frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,171 @@
+    });
+
+    it('Should receive the correct prop', async () => {
+        //default props
+        expect(wrapper.props()).toEqual(props);
+
+         await wrapper.setProps(newProps);
+         expect(wrapper.props()).toEqual(newProps);
+         expect(wrapper.props()).not.toEqual(props)
+    });
+
+    it('Should call and emit correct value' , async () => {
+        await usernameInputText.setValue(newValue);
+        emitChange = wrapper.emitted('change');
+        expect(wrapper.emitted('change')).toBeTruthy();
+        // test the given parameters when emitChange has been called
+        // i.e. emitChange = [ [ 'B' ] ]
+        expect(emitChange![0][0]).toEqual(newValue);
+
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult')
+        await wrapper.setProps(newProps);
+        await verifyButton.trigger('click');
+
+        expect(wrapper.emitted('setVerifyResult')).toBeTruthy();
+        expect(emitSetVerifyResult![1][0]).toEqual(true);
+    });
+
+    it('Should enable verify btn when username is inputted', async () => {
+        // button starts as disabled
+        expect((verifyButtonEl).disabled).toBe(true)
+        expect(verifyButton.classes('p-disabled')).toBe(true);
+
+        await wrapper.setProps({userId: newProps.userId })
+
+        expect((verifyButtonEl).disabled).toBe(false)
+        expect(verifyButton.classes('p-disabled')).toBe(false);
+    });
+
+    it('Should show loading on the verify btn while we call the api' , async () => {
+        vi.spyOn(
+            AppActlApiService.idirBceidProxyApi,
+            'idirSearch',
+        ).mockImplementation(async () => {
+            setLoadingState(true);
+            return userInputMock();
+        });
+
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toBe(newValue);
+
+        await wrapper.setProps({ userId: newProps.userId });
+        await verifyButton.trigger('click');
+        expect(isLoading()).toBe(true);
+        expect(verifyButtonEl.textContent).toContain('Loading');
+
+        // cleanup state variable
+        setLoadingState(false);
+        expect(isLoading()).toBe(false)
+    });
+
+    it('Should remove card and emit different value when domain changes', async () => {
+        emitSetVerifyResult = wrapper.emitted('setVerifyResult');
+        // default props
+        expect(wrapper.props()).toEqual(props);
+
+        await usernameInputText.setValue(newValue);
+        expect(usernameInputTextEl.value).toEqual(newValue);
+
+        await wrapper.setProps(userInputMock().data);
+        await verifyButton.trigger('click');
+        await flushPromises();
+
+        verifiedUserIdentity = wrapper.findComponent({ name: 'UserIdentityCard'}).vm;
+        verifiedUserIdentity = userInputMock().data
+        cardUsernameEl = wrapper.find('#userId').element as HTMLSpanElement;
+        expect(cardUsernameEl).toBeTruthy();
+        expect(cardUsernameEl.textContent).toContain('userId');
+
```

Comment:
We need another test for the user not found case.

Reply:
I'll add this test, thank you!